src/shogun/base/SGObject.h

```diff
@@ -315,6 +349,19 @@ class CSGObject
 	 */
 	virtual void save_serializable_post() throw (ShogunException);
 
+	virtual void fire_updated(int tag_id) 
+	{
+	};
+
+	template<typename T>
+	inline Tag<T> tag(const std::string& name) const
+	{
+	//	return m_context->tag_for<T>(name);
```

Comment:
@lisitsyn this was commented in aer so i left it like that for you to see. `tag_for` uses `anytag` of `Context` which was not implemented in aer. `Context` has `std::unordered_map<std::string, Any> m_tags;`  but i don't thing anything is getting stored in `m_tags` as of now. I think `anytag()` is supposed to query `m_tags`.